benchmarktests/target_secret_ldap.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package benchmarktests
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/hcl/v2/gohcl"
+	"github.com/hashicorp/vault/api"
+	vegeta "github.com/tsenart/vegeta/v12/lib"
+)
+
+// Constants for test
+const (
+	LDAPSecretTestType       = "ldap_secret"
+	LDAPSecretTestMethod     = "GET"
+	LDAPSecretBindPassEnvVar = VaultBenchmarkEnvVarPrefix + "LDAP_BIND_PASS"
+)
+
+func init() {
+	// "Register" this test to the main test registry
+	TestList[LDAPSecretTestType] = func() BenchmarkBuilder { return &LDAPSecretTest{} }
+}
+
+type LDAPSecretTest struct {
+	pathPrefix string
+	header     http.Header
+	roleName   string
+	config     *LDAPSecretTestConfig
+	logger     hclog.Logger
+}
+
+// Main Config Struct
+type LDAPSecretTestConfig struct {
+	LDAPConfig     *LDAPConfig     `hcl:"secret,block"`
+	LDAPRoleConfig *LDAPRoleConfig `hcl:"role,block"`
+}
+
+type LDAPConfig struct {
+	BindDN            string `hcl:"binddn"`
+	BindPass          string `hcl:"bindpass,optional"`
```

Comment:
is this optional?

Reply:
Yes and no.  It is optional as a parameter in the config, since it can be passed via env var

Reply:
got it, thank you!